server/char/list.py

```diff
@@ -102,38 +141,47 @@ def __init__(self, character_id, unknown1, character_name, character_unapproved_
 
 
     def serialize(self, stream):
-        stream.write(c_int64(character_id))
-        stream.write(c_uint32(unknown1))
-        stream.write(character_name)
-        stream.write(character_unapproved_name)
-        stream.write(is_name_rejected)
-        stream.write(free_to_play)
-        stream.write(unknown2, allocated_length=10)
-        stream.write(c_uint32(shirt_color))
-        stream.write(c_uint32(shirt_style))
-        stream.write(c_uint32(pants_color))
-        stream.write(c_uint32(hair_style))
-        stream.write(c_uint32(hair_color))
-        stream.write(c_uint32(lh))
-        stream.write(c_uint32(rh))
-        stream.write(c_uint32(eyebrows))
-        stream.write(c_uint32(eyes))
-        stream.write(c_uint32(mouth))
-        stream.write(c_uint32(unknown3))
-        stream.write(c_uint16(last_zone))
-        stream.write(c_uint16(last_instance))
-        stream.write(c_uint32(last_clone))
-        stream.write(c_uint64(last_login))
-        stream.write(c_uint16(len(items)))
-        for item in items:
+        """
+        Serializes the packet
+        """
+        stream.write(c_int64(self.character_id))
+        stream.write(c_uint32(self.unknown1))
+        stream.write(WString(self.character_name))
```

Comment:
You can use the `str` object, you just need to pass `allocated_length` to `stream.write`

Reply:
see comment above